sequencer/batchbuilder.go

```diff
@@ -69,11 +69,31 @@ func (s *Sequencer) tryToProcessTx(ctx context.Context, ticker *time.Ticker) {
 	for _, tx := range s.pendingTxs {
 		log.Infof("processing tx: %s", tx.Hash())
 	}
-	processedTxs, processedTxsHashes, unprocessedTxs, err := s.processTxs(ctx, s.pendingTxs)
+	processedTxs, processedTxsHashes, unprocessedTxs, unprocessedBatch, err := s.processTxs(ctx, s.pendingTxs)
 	if err != nil {
 		log.Errorf("failed to process txs, err: %w", err)
 		return
 	}
+
+	if unprocessedBatch {
+		// An Out Of Counter error has happened
+		// processedTxs must be reprocessed as the batch was discarded by the executor
+		reprocessTxs := make([]*pool.Transaction, len(processedTxs))
+		for _, pTxHash := range processedTxsHashes {
+			tx := s.getPendingTxByHash(s.pendingTxs, pTxHash)
+			reprocessTxs = append(reprocessTxs, tx)
+		}
+		reprocessedTxs, _, unreprocessedTxs, unreprocessedBatch, err := s.processTxs(ctx, reprocessTxs)
+		if err != nil {
+			log.Errorf("failed to reprocess txs, err: %w", err)
+			return
+		}
+		if len(reprocessedTxs) != len(reprocessTxs) || len(unreprocessedTxs) > 0 || unreprocessedBatch {
+			log.Error("failed to reprocess txs")
+			return
+		}
+	}
+
```

Comment:
Store the `unprocessedBatch` result of the first call in a separated var. Then close the batch if we've entered the for loop